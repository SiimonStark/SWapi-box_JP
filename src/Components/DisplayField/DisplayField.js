import React from 'react';
import Loading from '../Loading/Loading';
import Crawl from '../Crawl/Crawl';
import Card from '../Card/Card';
import PropTypes from 'prop-types';



let DisplayField = ({isLoading, isTab, currentFilm, person}) => {
  let determineTab = () => {
    switch (isTab) {
      case 'Crawl': return (<Crawl film={currentFilm} />);
      case 'People': return (<Card person={'personName'} />)
      default: console.log('nothing');
    }
  }
  let loading;
  let display;

  if (isLoading) {
    loading = <Loading />;
    display = null;
  } else {
    loading = null;
    display = determineTab();
  }

  return (
    <section className="DisplayField">
      {loading}
      {display}
    </section>
  )
}

DisplayField.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  isTab: PropTypes.string.isRequired,
  currentFilm: PropTypes.object.isRequired,
  person: PropTypes.array
}

export default DisplayField;