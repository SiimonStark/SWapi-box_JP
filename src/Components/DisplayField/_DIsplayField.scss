.DisplayField {
  border: white double 2px;
}