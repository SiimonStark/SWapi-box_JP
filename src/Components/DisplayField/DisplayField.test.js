import React from 'react';
import {mount} from 'enzyme';
import DisplayField from './DisplayField';
import {mockCrawl} from '../MockTestData';

describe('DisplayField', ()=>{
  let wrapper;
  let mockIsLoading = true;
  let mockIsTab = 'Crawl';
  let mockCurrentFilm = {mockCrawl};
  let mockPerson = 'personName';

  beforeEach(()=>{
    wrapper = mount(
      <DisplayField isLoading={mockIsLoading}
        isTab={mockIsTab} currentFilm={mockCurrentFilm}
        person={mockPerson} />
    );
  });

  it('should match snapshot if Loading', ()=>{
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should match snapshot if Not Loading', ()=>{
    wrapper = mount(
      <DisplayField isLoading={false} isTab={mockIsTab}
        currentFilm={mockCurrentFilm} person={mockPerson} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should change display based on isTab', ()=>{
    wrapper = mount (
      <DisplayField isLoading={false} isTab={'People'}
      currentFilm={mockCurrentFilm} person={mockPerson} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});