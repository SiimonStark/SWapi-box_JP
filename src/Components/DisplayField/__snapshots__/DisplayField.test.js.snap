// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayField should change display based on isTab 1`] = `
<DisplayField
  currentFilm={
    Object {
      "mockCrawl": Object {
        "circa": 1977,
        "crawl": "It is a period of civil war. Rebel spaceships, striking from a hidden base, have won their first victory against the evil Galactic Empire. During the battle, Rebel spies managed to steal secret plans to the Empire's ultimate weapon, the DEATH STAR, an armored space station with enough power to destroy an entire planet. Pursued by the Empire's sinister agents, Princess Leia races home aboard her starship, custodian of the stolen plans that can save her people and restore freedom to the galaxy.....",
        "title": "Episode IV - A New Hope",
      },
    }
  }
  isLoading={false}
  isTab="People"
  person="personName"
>
  <section
    className="DisplayField"
  >
    <Card
      person="personName"
    >
      <section
        className="card"
      >
        <div
          className="fade"
        />
        <h4>
          props.person
        </h4>
        <p>
          content
        </p>
      </section>
    </Card>
  </section>
</DisplayField>
`;

exports[`DisplayField should match snapshot if Loading 1`] = `
<DisplayField
  currentFilm={
    Object {
      "mockCrawl": Object {
        "circa": 1977,
        "crawl": "It is a period of civil war. Rebel spaceships, striking from a hidden base, have won their first victory against the evil Galactic Empire. During the battle, Rebel spies managed to steal secret plans to the Empire's ultimate weapon, the DEATH STAR, an armored space station with enough power to destroy an entire planet. Pursued by the Empire's sinister agents, Princess Leia races home aboard her starship, custodian of the stolen plans that can save her people and restore freedom to the galaxy.....",
        "title": "Episode IV - A New Hope",
      },
    }
  }
  isLoading={true}
  isTab="Crawl"
  person="personName"
>
  <section
    className="DisplayField"
  >
    <Loading>
      <div
        className="Loading"
      >
        <h3>
          Patience You Must
        </h3>
        <img
          alt="loader"
          src="Ripple-Loader.gif"
        />
      </div>
    </Loading>
  </section>
</DisplayField>
`;

exports[`DisplayField should match snapshot if Not Loading 1`] = `
<DisplayField
  currentFilm={
    Object {
      "mockCrawl": Object {
        "circa": 1977,
        "crawl": "It is a period of civil war. Rebel spaceships, striking from a hidden base, have won their first victory against the evil Galactic Empire. During the battle, Rebel spies managed to steal secret plans to the Empire's ultimate weapon, the DEATH STAR, an armored space station with enough power to destroy an entire planet. Pursued by the Empire's sinister agents, Princess Leia races home aboard her starship, custodian of the stolen plans that can save her people and restore freedom to the galaxy.....",
        "title": "Episode IV - A New Hope",
      },
    }
  }
  isLoading={false}
  isTab="Crawl"
  person="personName"
>
  <section
    className="DisplayField"
  >
    <Crawl
      film={
        Object {
          "mockCrawl": Object {
            "circa": 1977,
            "crawl": "It is a period of civil war. Rebel spaceships, striking from a hidden base, have won their first victory against the evil Galactic Empire. During the battle, Rebel spies managed to steal secret plans to the Empire's ultimate weapon, the DEATH STAR, an armored space station with enough power to destroy an entire planet. Pursued by the Empire's sinister agents, Princess Leia races home aboard her starship, custodian of the stolen plans that can save her people and restore freedom to the galaxy.....",
            "title": "Episode IV - A New Hope",
          },
        }
      }
    >
      <div
        className="Crawl"
      >
        <div
          className="fade"
        />
        <section
          className="Crawl--container"
        >
          <article
            className="Crawl--content"
          >
            <p />
            <h3>
               (
              )
            </h3>
          </article>
        </section>
      </div>
    </Crawl>
  </section>
</DisplayField>
`;
